lib/backup_helper.py

```diff
@@ -0,0 +1,44 @@
+import tarfile
+from os import path, getenv, remove
+from shutil import rmtree
+import sh
+
+directories = ['.screenly', 'screenly_assets']
+archive_name = "screenly-backup.tar.gz"
+
+
+def create_backup():
+    home = getenv('HOME')
+    file_path = path.join('static', archive_name)
+    if path.isfile(file_path):
+        remove(file_path)
+
+    try:
+        with tarfile.open(file_path, "w:gz") as tar:
+
+            for directory in directories:
+                path_to_dir = path.join(home, directory)
+                tar.add(path_to_dir, arcname=directory)
+    except IOError as e:
+        remove(file_path)
+        raise e
+
+    return archive_name
+
+
+def recover(file_path):
+    home = getenv('HOME')
+
+    with tarfile.open(file_path, "r:gz") as tar:
+        for directory in directories:
+            if directory not in tar.getnames():
+                raise Exception("Archive is wrong.")
+
+        sh.sudo('service', 'screenly-viewer', 'stop')
+        for directory in directories:
+            rmtree(path.join(home, directory))
+
+        tar.extractall(path=path.join(home))
+        sh.sudo('service', 'screenly-viewer', 'start')
```

Comment:
Sudo w/o passwd example from my ScreenBacklight script @ https://help.ubuntu.com/community/ScreenBacklight:

`
$ sudo visudo -f /etc/sudoers.d/myOverrides
`

Then copy/paste the following script into your new file and customize it:

`
%sudo	ALL=NOPASSWD:/usr/local/bin/sysbacklight
`

Make the script read-only:

`
$ sudo chmod 440 /etc/sudoers.d/myOverrides
`

Hope this helps.